pkg/machinery/config/provider.go

```diff
@@ -422,3 +423,8 @@ type InlineManifest interface {
 	Name() string
 	Contents() string
 }
+
+// UdevConfig describes configuration for udev
+type UdevConfig interface {
+	Rules() []string
+}
```

Comment:
Currently I'm going the route of just a single list of strings as rules, which will all be combined to a single rules file. Something like:

```yaml
udev:
  rules:
    - SUBSYSTEM=="drm", KERNEL=="card*", GROUP="44", MODE="0660"
    - SUBSYSTEM=="drm", KERNEL=="renderD*", GROUP="44", MODE="0660"
```

If preferred, we could do something more like this which would take a little more effort (and probably makes it a little more difficult for idempotency), but would produce multiple files:

```yaml
udev:
  rules:
    - name: 99-intel-gpu
      rules:
        - SUBSYSTEM=="drm", KERNEL=="card*", GROUP="44", MODE="0660"
        - SUBSYSTEM=="drm", KERNEL=="renderD*", GROUP="44", MODE="0660"
```